File: mobile/app/src/main/java/com/uniandes/abcall/data/model/Incident.kt
package com.uniandes.abcall.data.model

import androidx.room.Entity
import androidx.room.PrimaryKey
import java.io.Serializable
import java.sql.Timestamp

@Entity(tableName = "incidents_table")
data class Incident (
    @PrimaryKey
    val id:Int,
    val userId:String,
    val subject:String?,
    val description:String,
    val originType:String,
    val status:String,
    val solution:String?,
    val creationDate: Timestamp,
    val updateDate:Timestamp,
    val solutionAgentId:String?,
    val solutionDate:Timestamp?
): Serializable

File: frontend/abcall-web/src/app/user/incident.ts
export class Incident {
    subject: string;
    description: string;
    status: string;
    originType: string;
    solution: string;
    
    constructor(subject: string, description: string, status: string, originType: string, solution: string) {
        this.subject = subject;
        this.description = description;
        this.status = status;
        this.originType = originType;
        this.solution = solution;
    }
    
}
